### ISSUE TITLE
rmlint.sh: Generated script uses bash features, use "#!/bin/bash"

### ISSUE THREAD

**Issue by alexbarton** [sahib/rmlint#231]
The generated shell script (`-o:sh`) uses `bash` features, for example flags of the `echo` command which are undefined in POSIX shells, so it should use the correct shebang line:

```
#!/bin/bash
```

Without this, running the script on systems where `/bin/sh` is not `bash` results in garbled output (for example on macOS):

```
-n
[   0%] -n
 Keeping:   /tmp/test/a.jpg
-n
[  50%] -n
 Deleting:  /tmp/test/b.jpg
-n
[ 100%] -n
 Done!
Deleting script  /tmp/rmlint.sh

```

**Comment by alexbarton** [comment]
This is related to #28 and can probably fixed by using `printf` instead of `echo` (and keeping the `#!/bin/sh` shebang line) …

**Comment by hungrywolf27** [comment]
Similar issue on Ubuntu: running `rmlint.sh` produces this output, with progress always at 0%:
 
``` Keeping:   [filename omitted]
[   0%] ./rmlint.sh: 35: ./rmlint.sh: PROGRESS_CURR++: not found
 Deleting:  [filename omitted]
[   0%] ./rmlint.sh: 35: ./rmlint.sh: PROGRESS_CURR++: not found
 Keeping:   [filename omitted]
[   0%] ./rmlint.sh: 35: ./rmlint.sh: PROGRESS_CURR++: not found
```
(etc.)

Manually changing the shebang line to `#!/bin/bash` makes the progress work correctly.

**Comment by sahib** [comment]
Oww, those are some embarassing mistakes. I just noticed that ``/bin/sh`` indeed links to ``bash`` for me, 
probably part of the reason why I didn't notice this earlier.

6bd739b should fix most of this, however I need some more testing (and test cases) are needed.
The progress print function looks like this now: 

```bash
 print_progress_prefix() {
    printf "$COL_BLUE[% 4d%%]$COL_RESET " $((PROGRESS_CURR * 100 / PROGRESS_TOTAL))
    PROGRESS_CURR=$((PROGRESS_CURR+1))
 }
```

This seems to work with both ``dash`` and ``bash``. Portable shell programming is really a PITA.
I also checked if ``checkbashisms`` would complain, but seems to be fine for now.
